src/app/action/sigupActions.ts

```diff
@@ -0,0 +1,27 @@
+'use server';
+
+import { SignUpFormValues } from '@/app/components/Forms/Forms.types';
+
+export const signupAction = async (data: SignUpFormValues) => {
+  try {
+    const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/auth/signup`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (!response.ok) {
+      const error = await response.json();
+      console.error('Signup failed:', error);
+      throw new Error(error.message || 'Signup failed');
```

Comment:
nitpick: I noticed we’re both logging the error with console.error and then throwing a new Error with the same message. It might be cleaner to just throw and let the caller handle logging. This would reduce redundant logs and keep the code concise. What do you think?
```suggestion
const error = await response.json();
throw new Error(error.message || 'Signup failed');
```